### ISSUE TITLE
[DEV] Slow walking speed

### ISSUE THREAD

**Issue by mrseeker** [PokemonGoF/PokemonGo-Bot#1586]
### Expected Behavior
Bot walks like a normal human

### Actual Behavior
```
[05:23:52] Moving towards fort 41fa8df35089444994e59fcadcd3d080.16, 0.04km left
[05:24:42] Moving towards fort 41fa8df35089444994e59fcadcd3d080.16, 0.04km left
[05:25:32] Moving towards fort 41fa8df35089444994e59fcadcd3d080.16, 0.03km left
[05:26:22] Moving towards fort 41fa8df35089444994e59fcadcd3d080.16, 0.03km left
[05:27:13] Moving towards fort 41fa8df35089444994e59fcadcd3d080.16, 0.02km left
[05:28:02] Moving towards fort 41fa8df35089444994e59fcadcd3d080.16, 0.02km left
```
Which I assume is crawling (0,468km/h)...

### Steps to Reproduce
From the config:
```
"walk": 4.46,
"distance_unit": "km",
```

### Other Information
OS: Ubuntu 14.02
Git Commit: a1433f235c0afc00a15d79ff99711c0cd520dc7f

**Comment by Jojo1910** [comment]
If you set walk to 4.46 it will walk about 4km/h you need to set it higher

**Comment by mrseeker** [comment]
Okay, let me calculate something because I think that the calculations are wrong then:
At 05:24:42 it was at 0.04km = 40 meters.
At 05:27:13 it was at 0.02km = 20 meters.

This means that the bot walks 0.02km (or 20 meters) in 51 seconds, that is 0.13m/s or 0,468km/h.

Might need an extra 0 to match the speed???

**Comment by urbels** [comment]
The same for me since this mornings dev updates.

**Comment by arnogues** [comment]
4km/h is human speed, put it to 13 or 14 (max or the egg never eclore)

**Comment by mrseeker** [comment]
@arnogues I changed my initial post to match the actual speed that I get out of it, which is 0,468km/h. Unless I am mistaken, that is the speed that I am seeing here right now and is 10% of the speed I set in the configuration...

**Comment by klingan** [comment]
Yep, same issue, last dev broke the speed.

**Comment by Bashin** [comment]
#1570 
#1574

**Comment by tstumm** [comment]
Distance unit it only for displaying distances. It does NOT affect walking speed in any way.

**Comment by tstumm** [comment]
The bot is walking so slow because on each tick we only walk `walk` meters from the current location. But the tick is way longer than just 1 second, so we actually slow down.

**Comment by mrseeker** [comment]
Yeah, I just noticed that the speed is a bit irregular, might be handy to add the actual speed it travels...

**Comment by tstumm** [comment]
Also the README is wrong because someone merged wrong information.

**Comment by mrseeker** [comment]
Suggestion: to avoid confusion: Make the actual speed of the bot more visible.

[06:03:09] Walking from (52.381166, 4.884247) to [52.382961202506465, 4.884675595425898] 0.20km _@ 10km/h_

**Comment by mrthrowaway** [comment]
> [15:41:33] Ran for 0:29:47
> [15:41:33] Travelled 0.05km

This was at "walk" 10.25 on dev

It was moving at a lot more reasonable speed yesterday (15~ hours ago) at the default 4.16

**Comment by klingan** [comment]
[12:11:31] Moving towards fort 42fb0fba44654ddfbfc5804516b49072.16, 0.06km left [                                 ] 0%
[12:12:42] Moving towards fort 42fb0fba44654ddfbfc5804516b49072.16, 0.05km left [|||||                            ] 16%
[12:13:54] Moving towards fort 42fb0fba44654ddfbfc5804516b49072.16, 0.04km left [|||||||||||                      ] 33%
[12:15:06] Moving towards fort 42fb0fba44654ddfbfc5804516b49072.16, 0.03km left [||||||||||||||||                 ] 49%

config:    "walk": 9.2,


seems to meters/minute

**Comment by mrseeker** [comment]
```
[06:14:50] Ran for 0:13:45
[06:14:50] Travelled 0.13km
```
0.13/0.23h = 0.56km/h (for setting = 10)

**Comment by arnogues** [comment]
```
[12:20:27] Ran for 0:11:13
[12:20:27] Total XP Earned: 1950  Average: 10432.23/h
[12:20:27] Travelled 0.23km
```

```
"walk": 50, 
```

**Comment by tstumm** [comment]
I guess you all get the point. Looking for pokemon and stuff just takes time. We can't really change that. Increasing `walk` just leads to walking past pokemons without catching

**Comment by arnogues** [comment]
The trainer walk is very slow and we don't understand why

**Comment by Bashin** [comment]
Setting "evolve_all" from "all" to "none" does the trick for me -> bot is walking like normal again.
Maybe its not a good idea to try to evolve every pokemon on every step...

**Comment by klingan** [comment]
@Bashin right, i just notice the same when increasing the evolve_speed value.

**Comment by mrthrowaway** [comment]
@Bashin can confirm. setting it to 'none' makes it go at the normal speed

**Comment by klingan** [comment]
To fix: increase evolve_speed to 200 instead of 20

**Comment by BoGnY** [comment]
I confirm problem..
Bot only walk.. Don't do anything..
It walk near pokestop and do nothing..

**Comment by mrseeker** [comment]
At speed = 10 and evolve_speed = 200

```
[06:42:21] [x] Next egg incubates in 0.83 km
[06:45:54] [x] Next egg incubates in 0.75 km
[06:50:25] [x] Next egg incubates in 0.70 km
[06:54:49] [x] Next egg incubates in 0.65 km
```
213 seconds to walk 0.08km, which is 1.35km/h
271 seconds to walk 0.05km, which is 0.66km/h
264 seconds to talk 0.05km, which is 0.68km/h

Note here that I get a lot of pokemon, so it might potentially be true that the speed is slow because of all the pokemon... However, I see a pattern that it is 10% of the average speed of 10km/h.

**Comment by Bashin** [comment]
I set walk to 10 and get about 9km/h when it is moving to a pokestop which is pretty good. Evolve all is set to none though

**Comment by mrseeker** [comment]
That might be the cause of the issue: I set evolve_all = none and suddenly the speed increases to its normal potential...

**Comment by Bashin** [comment]
That's what I was writing 35 minutes ago ;-)

**Comment by mrseeker** [comment]
I noticed that one, I just wanted to be sure that it was related to evolve and not some random factor...

**Comment by klingan** [comment]
@mrseeker right, sorry about that, i saw my evolve was still off from the first test :dizzy_face:

**Comment by mrseeker** [comment]
I found the bug, but need someone to verify this and push this to dev:

https://github.com/PokemonGoF/PokemonGo-Bot/blob/dev/pokemongo_bot/cell_workers/evolve_all_worker.py#L61
```
# Evolve all is used - Don't run after the first tick or if the config flag is false
        if self.bot.tick_count is not 0 or not self.config.use_lucky_egg:
            return **True**
```
must be **False**

**Comment by BoGnY** [comment]
don't work for me.. bot walks always very very slow :(

**Comment by bnt44** [comment]
1. Change line 61 to False https://github.com/PokemonGoF/PokemonGo-Bot/blob/dev/pokemongo_bot/cell_workers/evolve_all_worker.py#L61

In config file: 
```
"evolve_all": "NONE",
"evolve_speed": 200,
"use_lucky_egg": false,
"evolve_captured": "NONE",
```

this working for me 

> [16:51:52] Moving towards fort xxx, 0.16km left
> [16:51:56] 
> [16:51:57] Moving towards fort xxx, 0.15km left
> [16:52:00] 
> [16:52:01] Moving towards fort xxx, 0.14km left
> [16:52:04] 
> [16:52:06] Moving towards fort xxx, 0.12km left
> [16:52:09] 
> [16:52:10] Moving towards fort xxx, 0.11km left

0,05km in 18 seconds = 10km/h 
in config file i have 12.3km/h

**Comment by mrseeker** [comment]
If someone can push a patch, that would be great, since I am @ work and cannot create a patch.

**Comment by Erody** [comment]
None of this is working for me. Still walking at an abnormally slow pace.   
   
`[18:01:10] Moving towards fort 3aabee0e7fa040c8a50b3b4571fdb19c.16, 0.08km left
[18:01:14] 
[18:01:15] Moving towards fort 3aabee0e7fa040c8a50b3b4571fdb19c.16, 0.07km left
[18:01:19] 
[18:01:20] Moving towards fort 3aabee0e7fa040c8a50b3b4571fdb19c.16, 0.07km left
[18:01:24] 
[18:01:25] Moving towards fort 3aabee0e7fa040c8a50b3b4571fdb19c.16, 0.06km left
[18:01:29] 
[18:01:30] Moving towards fort 3aabee0e7fa040c8a50b3b4571fdb19c.16, 0.06km left
[18:01:34] 
[18:01:36] Moving towards fort 3aabee0e7fa040c8a50b3b4571fdb19c.16, 0.05km left
[18:01:39] 
[18:01:40] [x] Next egg incubates in 2.95 km
[18:01:41] Moving towards fort 3aabee0e7fa040c8a50b3b4571fdb19c.16, 0.05km left
[18:01:44] 
[18:01:45] Moving towards fort 3aabee0e7fa040c8a50b3b4571fdb19c.16, 0.05km left
[18:01:49] 
[18:01:51] Moving towards fort 3aabee0e7fa040c8a50b3b4571fdb19c.16, 0.04km left
[18:01:54] 
[18:01:55] Moving towards fort 3aabee0e7fa040c8a50b3b4571fdb19c.16, 0.04km left`

**Comment by mrseeker** [comment]
Erody, what version are you using? Latest dev or not? You need to alter the file to make it work, they have not pushed a patch out yet.

**Comment by vamirineni87** [comment]
seem like there is no control on the merges. Things are all over the place past few days. We should start pushing stable releases.

**Comment by mrseeker** [comment]
Seems to have been fixed with pull #1774

**Comment by mrthrowaway** [comment]
Cannot confirm. Changing back to evolve_all "all" at 7.16km walk speed gives me this

[15:09:15] Ran for 0:15:08
[15:09:15] Total XP Earned: 160  Average: 634.57/h
[15:09:15] Travelled 0.01km
[15:09:15] Visited 1 stops

Same config with evolve_all "NONE" walks at expected speed
